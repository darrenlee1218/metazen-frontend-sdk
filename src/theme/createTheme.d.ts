import { Theme as SystemTheme, ThemeOptions } from '@mui/system';
import { CustomTheme, Palette } from '@mui/material/styles';

declare module '@mui/material/styles' {
  export interface ColorsInterface {
    primary: string;
    appBackground: string;
    inputBackground: string;
    bottomNavBackground: string;
    closeButtonBackground: string;
    primaryText: string;
    secondaryText: string;
    placeholderText: string;
    clickableGray: string;
    border: string;
    green: string;
    clickableGrayHover: string;
    hoverColor: string;
    disabledColor: string;
    nonClickableGray: string;
    loadingPrimary: string;
    loadingSecondary: string;
    errorPrimary: string;
    errorSecondary: string;
    successPrimary: string;
    successSecondary: string;
    warningPrimary: string;
    warningSecondary: string;
  }

  interface CustomPalette extends Palette {
    colors: ColorsInterface;
  }

  interface CustomTheme extends SystemTheme {
    palette: CustomPalette;
  }
  // allow configuration using `createTheme`
  interface CustomThemeOptions extends ThemeOptions {
    palette?: CustomPalette;
  }

  export function createTheme(options?: CustomThemeOptions): CustomTheme;
}

declare module '@emotion/react' {
  interface Theme extends CustomTheme {}
}
