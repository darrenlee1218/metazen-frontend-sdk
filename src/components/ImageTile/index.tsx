import React from 'react';
import { Box } from '@mui/system';
import { CustomTheme, useTheme, CSSObject } from '@mui/material';

interface ImageProps {
  src: string;
  sx?: CSSObject;
}

export default function ImageTile({ src = '', sx = {} }: ImageProps) {
  const theme = useTheme() as CustomTheme;
  return (
    <Box
      component="div"
      bgcolor={theme.palette.colors.clickableGray}
      sx={{
        width: '100%',
        minHeight: '20px',
        border: '1px solid',
        borderColor: theme.palette.colors.border,
        backgroundImage: `url(${src})`,
        backgroundSize: 'cover',
        backgroundRepeat: 'no-repeat',
        backgroundPosition: 'center',
        ...sx,
      }}
    />
  );
}
