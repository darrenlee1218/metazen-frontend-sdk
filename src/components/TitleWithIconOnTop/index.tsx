import React from 'react';
import Box from '@mui/material/Box';

import TitleWithIconOnTopProps from '@gryfyn-types/props/TitleWithIconOnTopProps';

const TitleWithIconOnTop = ({ imageLayout, titleLayout }: TitleWithIconOnTopProps): React.ReactElement => {
  return (
    <Box
      sx={{
        display: 'flex',
        flex: 1,
        alignItems: 'center',
        flexDirection: 'column',
      }}
    >
      {imageLayout}
      {titleLayout}
    </Box>
  );
};

export default TitleWithIconOnTop;
