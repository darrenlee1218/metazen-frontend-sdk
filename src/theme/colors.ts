import { ColorsInterface } from '@mui/material/styles';

const Colors: ColorsInterface = {
  // game/animoca brands primary color(s)
  // background
  appBackground: '#111619',
  inputBackground: '#0d1014',
  bottomNavBackground: 'rgba(55, 71, 80, 0.55)',
  closeButtonBackground: 'rgba(255, 255, 255, 0.15)',
  // Text
  primaryText: '#ffffff',
  secondaryText: '#8A9296',
  placeholderText: '#47505D',
  // others
  clickableGray: '#20282D',
  border: '#20282d',
  green: '#67d65b',
  clickableGrayHover: '#303B43',
  primary: '#12ABEC',
  hoverColor: '#3BBBF1',
  disabledColor: '#04394F',
  nonClickableGray: '#171D21',

  loadingPrimary: '#42CCEC',
  loadingSecondary: '#0A2930',
  errorPrimary: '#FA4D56',
  errorSecondary: '#2C0608',
  successPrimary: '#67D65B',
  successSecondary: '#0E250C',
  warningPrimary: '#EB7C4A',
  warningSecondary: '#20140E',
};

export default Colors;
