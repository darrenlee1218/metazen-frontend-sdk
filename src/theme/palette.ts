import { DeepPartial } from '@gryfyn-types/generics';
import { CustomPalette } from '@mui/material/styles';
import Colors from './colors';
import { Mode } from './types';

export const commonPallete = {
  mode: Mode.Dark,
  primary: {
    main: Colors.primary,
  },
  colors: {
    ...Colors,
  },
  text: {
    primary: Colors.primaryText,
    secondary: Colors.secondaryText,
  },
  background: {
    default: Colors.appBackground,
    paper: Colors.appBackground,
  },
  loading: {
    main: Colors.loadingPrimary,
  },
  error: {
    main: Colors.errorPrimary,
  },
  info: {
    main: '#42CCEC',
  },
  success: {
    main: Colors.successPrimary,
  },
  warning: {
    main: Colors.warningPrimary,
  },
};

export type AdditionalPalette = DeepPartial<CustomPalette>;

const palette = (additionalPalette?: AdditionalPalette) =>
  ({
    ...commonPallete,
    ...additionalPalette,
    colors: {
      ...commonPallete.colors,
      ...additionalPalette?.colors,
    },
  } as unknown as CustomPalette);

export default palette;
