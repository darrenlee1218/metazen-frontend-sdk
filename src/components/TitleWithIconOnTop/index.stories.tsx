import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import Typography from '@mui/material/Typography';

import ImageTile from '../../../stateless/atoms/ImageTile';
import TitleWithIconOnTop from '.';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'common-components/stateless/organisms/TitleWithIconOnTop',
  component: TitleWithIconOnTop,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {},
} as ComponentMeta<typeof TitleWithIconOnTop>;

const Template: ComponentStory<typeof TitleWithIconOnTop> = (args) => <TitleWithIconOnTop {...args} />;

const mockGameImgSrc = 'https://picsum.photos/200';

const imageLayout = (
  <ImageTile
    src={mockGameImgSrc}
    sx={{
      maxWidth: '60px',
      borderRadius: '8px',
      minHeight: '60px',
    }}
  />
);

const titleLayout = (
  <>
    <Typography variant="h1" color="text.primary" sx={{ fontWeight: 'bold' }}>
      Signature Request
    </Typography>
    <Typography variant="h5" color="text.primary">
      url-of-game.com
    </Typography>
  </>
);

export const Empty = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Empty.args = {
  imageLayout,
  titleLayout,
};

export const Default = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {
  imageLayout,
  titleLayout,
};
