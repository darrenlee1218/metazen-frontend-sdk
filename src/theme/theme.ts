import { createTheme } from '@mui/material/styles';
import { ButtonProps } from '@mui/material/Button';
import palette, { AdditionalPalette } from './palette';
import colors from './colors';

export const defaultTheme = (additionalPalette?: AdditionalPalette) =>
  createTheme({
    palette: palette(additionalPalette),
    typography: {
      fontFamily: ['Basier Circle', 'graphie'].join(','),
      h6: {
        fontSize: '8px',
      },
      h5: {
        fontSize: '10px',
      },
      h4: {
        fontSize: '12px',
      },
      h3: {
        fontSize: '14px',
      },
      h2: {
        fontSize: '16px',
      },
      h1: {
        fontSize: '24px',
      },
    },
    shape: {
      borderRadius: 1,
    },
    breakpoints: {
      values: {
        xs: 0,
        sm: 600,
        md: 900,
        lg: 1200,
        xl: 1536,
      },
    },
    components: {
      MuiCssBaseline: {
        styleOverrides: `
        @font-face {
          font-family: 'Basier Circle';
          font-style: normal;
          font-display: swap;
          font-weight: 400;
          src: local('Basier Circle Regular'), local('BasierCircle-Regular'), url('./fonts/BasierCircle-Regular.ttf') format('truetype');
          unicodeRange: 'U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF';
        }

        @font-face {
          font-family: 'Basier Circle';
          font-style: italic;
          font-display: swap;
          font-weight: 400;
          src: local('Basier Circle Regular Italic'), local('BasierCircle-RegularItalic'), url('./fonts/BasierCircle-RegularItalic.ttf') format('truetype');
          unicodeRange: 'U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF';
        }

        @font-face {
          font-family: 'Basier Circle';
          font-style: normal;
          font-display: swap;
          font-weight: 500;
          src: local('Basier Circle Medium'), local('BasierCircle-Medium'), url('./fonts/BasierCircle-Medium.ttf') format('truetype');
          unicodeRange: 'U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF';
        }

        @font-face {
          font-family: 'Basier Circle';
          font-style: italic;
          font-display: swap;
          font-weight: 500;
          src: local('Basier Circle Medium Italic'), local('BasierCircle-MediumItalic'), url('./fonts/BasierCircle-MediumItalic.ttf') format('truetype');
          unicodeRange: 'U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF';
        }

        @font-face {
          font-family: 'Basier Circle';
          font-style: normal;
          font-display: swap;
          font-weight: 600;
          src: local('Basier Circle SemiBold'), local('BasierCircle-SemiBold'), url('./fonts/BasierCircle-SemiBold.ttf') format('truetype');
          unicodeRange: 'U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF';
        }

        @font-face {
          font-family: 'Basier Circle';
          font-style: italic;
          font-display: swap;
          font-weight: 600;
          src: local('Basier Circle SemiBold Italic'), local('BasierCircle-SemiBoldItalic'), url('./fonts/BasierCircle-SemiBoldItalic.ttf') format('truetype');
          unicodeRange: 'U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF';
        }

        @font-face {
          font-family: 'Basier Circle';
          font-style: normal;
          font-display: swap;
          font-weight: 700;
          src: local('Basier Circle Bold'), local('BasierCircle-Bold'), url('./fonts/BasierCircle-Bold.ttf') format('truetype');
          unicodeRange: 'U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF';
        }

        @font-face {
          font-family: 'Basier Circle';
          font-style: italic;
          font-display: swap;
          font-weight: 700;
          src: local('Basier Circle Bold Italic'), local('BasierCircle-BoldItalic'), url('./fonts/BasierCircle-BoldItalic.ttf') format('truetype');
          unicodeRange: 'U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF';
        }
        input:-webkit-autofill,
        input:-webkit-autofill:focus {
          transition: background-color 600000s 0s, color 600000s 0s;
          box-shadow: none;
        }
      `,
      },
      MuiMobileStepper: {
        styleOverrides: {
          dot: {
            backgroundColor: colors.placeholderText,
          },
          dotActive: {
            backgroundColor: colors.primaryText,
          },
        },
      },
      MuiOutlinedInput: {
        styleOverrides: {
          root: {
            backgroundColor: colors.inputBackground,
            borderRadius: 4,
            input: {
              fontSize: 12,
              '&::placeholder': {
                opacity: 1,
              },
              '&:-webkit-autofill': {
                '-webkitBoxShadow': `0 0 0px 1000px ${colors.inputBackground} inset`,
                '-webkit-text-fill-color': colors.primaryText,
              },
              '&:-webkit-autofill:focus': {
                '-webkitBoxShadow': `0 0 0px 1000px ${colors.inputBackground} inset`,
                '-webkit-text-fill-color': colors.primaryText,
              },
            },
            '&:hover': {
              border: '#282E36',
            },
            'input::placeholder': {
              color: colors.placeholderText,
            },
          },
          notchedOutline: {
            borderWidth: '1px !important',
            borderColor: `${additionalPalette?.colors?.border ?? colors.border} !important`,
            '.Mui-focused &': {
              borderColor: 'white !important',
            },
            '.MuiInputBase-colorWarning &': {
              borderColor: `${additionalPalette?.colors?.warningPrimary ?? colors.warningPrimary} !important`,
            },
            '.Mui-error &': {
              borderColor: `${additionalPalette?.colors?.errorPrimary ?? colors.errorPrimary} !important`,
            },
          },
        },
      },
      MuiInputLabel: {
        styleOverrides: {
          root: {
            'input ~ svg': {
              color: additionalPalette?.colors?.border ?? colors.border,
            },
            'input:checked ~ svg': {
              color: additionalPalette?.primary?.main ?? colors.primary,
            },
          },
        },
      },
      MuiButton: {
        styleOverrides: {
          // eslint-disable-next-line complexity
          root: ({ ownerState }: { ownerState: ButtonProps }) => ({
            textTransform: 'none',
            ...(ownerState.color === 'primary' && {
              color: colors.primaryText,
              '&:disabled': {
                backgroundColor: additionalPalette?.colors?.disabledColor ?? colors.disabledColor,
              },
            }),
            ...(ownerState.color === 'primary' &&
              ownerState.variant === 'contained' && {
                color: colors.primaryText,
                '&:hover': {
                  backgroundColor: additionalPalette?.colors?.hoverColor ?? colors.hoverColor,
                },
                '&:disabled': {
                  backgroundColor: additionalPalette?.colors?.disabledColor ?? colors.disabledColor,
                },
              }),
            ...(ownerState.color === 'secondary' && {
              color: colors.primaryText,
              borderColor: colors.secondaryText,
              backgroundColor: 'inherit',
              '&:hover': {
                borderColor: colors.secondaryText,
                backgroundColor: 'inherit',
              },
            }),
            ...(ownerState.color === 'secondary' &&
              ownerState.variant === 'contained' && {
                backgroundColor: colors.clickableGray,
                '&:hover': {
                  borderColor: colors.secondaryText,
                  backgroundColor: colors.clickableGrayHover,
                },
              }),
          }),
        },
      },
      MuiIconButton: {
        styleOverrides: {
          root: {
            backgroundColor: colors.clickableGray,
            '&.Mui-disabled': {
              backgroundColor: colors.clickableGray,
              opacity: '50%',
            },
          },
        },
      },
      MuiTooltip: {
        styleOverrides: {
          arrow: {
            color: colors.secondaryText,
          },
          tooltip: {
            backgroundColor: colors.secondaryText,
            padding: '4px',
            fontSize: 10,
            borderRadius: 3,
            fontWeight: 400,
          },
          tooltipPlacementTop: {
            marginBottom: '4px !important', // Hard to override tooltipPlacementTop
          },
        },
        defaultProps: {
          placement: 'top',
        },
      },
      MuiBottomNavigationAction: {
        styleOverrides: {
          root: {
            color: colors.secondaryText,
            '&.Mui-selected': {
              color: colors.primaryText,
              '.MuiTouchRipple-root': {
                height: '2px',
                width: '48px',
                marginLeft: 'auto',
                marginRight: 'auto',
                backgroundColor: additionalPalette?.primary?.main ?? colors.primary,
              },
            },
          },
        },
      },
      MuiTabs: {
        styleOverrides: {
          root: {
            '.MuiTabs-indicator': {
              height: '1px',
            },
          },
        },
      },
      MuiCardActionArea: {
        styleOverrides: {
          root: {
            ':hover': {
              backgroundColor: colors.clickableGrayHover,
            },
          },
        },
      },
      MuiFormHelperText: {
        styleOverrides: {
          root: {
            fontSize: '10px',
            '.MuiInputBase-colorWarning ~ &': {
              color: `${additionalPalette?.colors?.warningPrimary ?? colors.warningPrimary} !important`,
            },
          },
        },
      },
      MuiList: {
        styleOverrides: {
          root: {
            backgroundColor: colors.appBackground,
            padding: 8,
          },
        },
      },
      MuiMenuItem: {
        styleOverrides: {
          root: {
            height: '23px',
            minHeight: '23px',
            padding: '4px 8px',
            borderRadius: '3px',
            fontSize: 12,
            backgroundColor: colors.appBackground,
            '&:hover': {
              backgroundColor: colors.clickableGrayHover,
            },
            '&.Mui-selected': {
              backgroundColor: colors.nonClickableGray,
              color: colors.placeholderText,
              '&:hover': {
                backgroundColor: colors.nonClickableGray,
              },
            },
          },
        },
      },
    },
  });

export default defaultTheme;
