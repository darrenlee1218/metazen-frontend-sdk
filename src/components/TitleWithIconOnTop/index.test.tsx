import React from 'react';
import TestRenderer from 'react-test-renderer';
import Typography from '@mui/material/Typography';
import { ThemeProvider } from '@mui/material/styles';

import TitleWithIconOnTop from './index';
import ImageTile from '@components/ImageTile';
import theme from '@theme/theme';

const mockGameImgSrc = 'https://picsum.photos/200';

const imageLayout = (
  <ImageTile
    src={mockGameImgSrc}
    sx={{
      maxWidth: '60px',
      borderRadius: '8px',
      minHeight: '60px',
    }}
  />
);

const titleLayout = (
  <>
    <Typography variant="h1" color="text.primary" sx={{ fontWeight: 'bold' }}>
      Signature Request
    </Typography>
    <Typography variant="h5" color="text.primary">
      url-of-game.com
    </Typography>
  </>
);

describe('TitleWithIconOnTop', () => {
  test('renders given non-empty layout', () => {
    const recentActivity = TestRenderer.create(
      <ThemeProvider theme={theme()}>
        <TitleWithIconOnTop imageLayout={imageLayout} titleLayout={titleLayout} />
      </ThemeProvider>,
    );
    expect(recentActivity.toJSON()).toMatchSnapshot();
  });
});
