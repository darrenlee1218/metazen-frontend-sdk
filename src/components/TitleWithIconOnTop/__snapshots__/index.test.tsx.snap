// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleWithIconOnTop renders given non-empty layout 1`] = `
<div
  className="MuiBox-root css-mzfgik"
>
  <div
    className="MuiBox-root css-1upnith"
  />
  <h1
    className="MuiTypography-root MuiTypography-h1 css-yr8l96-MuiTypography-root"
  >
    Signature Request
  </h1>
  <h5
    className="MuiTypography-root MuiTypography-h5 css-14tnc7k-MuiTypography-root"
  >
    url-of-game.com
  </h5>
</div>
`;
